modules/images/image-loading-optimization/storage/post-type.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * Metrics storage post type.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+const ILO_PAGE_METRICS_POST_TYPE = 'ilo_page_metrics';
+
+/**
+ * Register post type for page metrics storage.
+ *
+ * This the configuration for this post type is similar to the oembed_cache in core.
+ */
+function ilo_register_page_metrics_post_type() /*: void (in PHP 7.1) */ {
+	register_post_type(
+		ILO_PAGE_METRICS_POST_TYPE,
+		array(
+			'labels'           => array(
+				'name'          => __( 'Page Metrics', 'performance-lab' ),
+				'singular_name' => __( 'Page Metrics', 'performance-lab' ),
+			),
+			'public'           => false,
+			'hierarchical'     => false,
+			'rewrite'          => false,
+			'query_var'        => false,
+			'delete_with_user' => false,
+			'can_export'       => false,
+			'supports'         => array( 'title' ), // The original URL is stored in the post_title, and the post_name is a hash of the query vars.
+		)
+	);
+}
+add_action( 'init', 'ilo_register_page_metrics_post_type' );
+
+/**
+ * Get page metrics post.
+ *
+ * @param string $slug Page metrics slug.
+ * @return WP_Post|null Post object if exists.
+ */
+function ilo_get_page_metrics_post( string $slug ) /*: ?WP_Post (in PHP 7.1) */ {
```

Comment:
why does this say "(in PHP 7.1)"?

Reply:
Because this is only syntactically valid in PHP 7.1 ([3v4l](https://3v4l.org/YGSZf)):

```
function ilo_get_page_metrics_post( string $slug ): ?WP_Post {
    // ...
}
```

Reply:
Maybe I went over the top with adding PHP type declarations, but in the future the comments could be removed when WP is bumped to that version of PHP.

Reply:
Ah, right - thanks.

Reply:
I'd say this is a bit overkill for now. There are a ton of places like this in core and in plugins, and I don't think it's worth adding this to all of them. So I would say let's stick with what we _can_ add at this point.